### ISSUE TITLE
GitHub with enableReleaseAssets: version issue

### ISSUE THREAD

**Issue by eteubert** [YahnisElsts/plugin-update-checker#370]
I am using v4.9 with a configuration like this:

```php
<?php
$myUpdateChecker = Puc_v4_Factory::buildUpdateChecker(
    'https://github.com/' . $plugin->github . '/',
    $plugin_base_path . $plugin->file,
    $plugin->slug
);
$myUpdateChecker->getVcsApi()->enableReleaseAssets('/' . $plugin->slug . '/');
```

It works _but_ it always shows an update, even when I'm on the latest version.

<img width="753" alt="Screenshot 2020-05-18 at 22 41 46" src="https://user-images.githubusercontent.com/235918/82257921-2d567000-9959-11ea-868d-b7ff673d5b39.png">

The `Version` header in this case is `v3.0-beta12` but the updater seems to compare to just `3.0`.

**Comment by YahnisElsts** [comment]
I'm not sure why that's happening. If your repository is public, could you post a link to the release that the update checker is reporting as an available update?

Also, what's the `Version` header of the currently installed version?

**Comment by eteubert** [comment]
This is the release: https://github.com/podlove/podlove-publisher/releases/tag/v3.0-beta12

The version header line is ` * Version: v3.0-beta12`

**Comment by YahnisElsts** [comment]
Thank you. I think I see the problem: even when you enable release assets, the update checker will only look at the plugin header in the relevant GitHub tag, not in the actual release asset (i.e. the ZIP file). And the header in the [`v3.0-beta12` tag](https://github.com/podlove/podlove-publisher/blob/v3.0-beta12/podlove.php#L6) says `Version:     3.0`.

In theory, we could fix this by changing the update checker to extract the plugin header from release assets. However, I don't think it's practical to download and parse multi-megabyte ZIP files every time a plugin checks for updates. 

In this case, I'd recommend making sure that the `Version` header in the release tag matches the `Version` header in the release asset.

**Comment by eteubert** [comment]
Ah, that makes sense. Thanks for the explanation and quick response 💜
